zipline/rl_manager.py

```diff
@@ -0,0 +1,111 @@
+import abc
+from six import with_metaclass
+from collections import namedtuple
+from intervaltree import IntervalTree, Interval
+from itertools import groupby
+
+
+Restriction = namedtuple(
+    'Restriction', ['sid',
+                    'effective_date',
+                    'expiry_date',
+                    'restriction_type']
+)
+
+
+class RestrictionsController(object):
+    """
+    Combines restriction information of various sources from multiple
+    restriction managers
+    """
+
+    def __init__(self):
+        self.rl_managers = []
+
+    def add_restrictions(self, rl_manager):
+        self.rl_managers.append(rl_manager)
+
+    def restrictions(self, sid, dt):
+        restrictions = [rlm.restrictions(sid, dt) for rlm in self.rl_managers]
+        return set.union(*restrictions) if restrictions else {}
+
+    def is_restricted(self, sid, dt):
+        return len(self.restrictions(sid, dt)) != 0
+
+
+class RLManager(with_metaclass(abc.ABCMeta)):
+    """
+    ABC for a restricted list manager that returns information about
+    restrictions from a single source
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def restrictions(self, sid, dt):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def is_restricted(self, sid, dt):
```

Comment:
This should probably take an iterable of sids to allow for a future vectorized implementation.